icevision/models/mmdet/common/bbox/single_stage/model.py

```diff
@@ -5,26 +5,49 @@
 from mmdet.models import build_detector
 from mmcv.runner import load_checkpoint
 from icevision.models.mmdet.utils import *
+from icevision.utils.download_utils import *
 
 
 def model(
-    cfg: Union[str, Path, Config],
+    backbone,
     num_classes: int,
-    weights_path: Optional[Union[str, Path]] = None,
+    pretrained: bool = True,
+    checkpoints_path: Optional[Union[str, Path]] = None,
+    force_download=False,
 ) -> nn.Module:
 
-    # if `cfg` argument is a path (str, Path) create an Config object from the file
-    # otherwise cfg should be already an Config object
-    if isinstance(cfg, (str, Path)):
-        cfg = Config.fromfile(str(cfg))
+    save_dir = None
+
+    model_name = backbone["model_name"]
+    cfg_filepath = backbone["cfg_filepath"]
+    weights_url = backbone["weights_url"]
+
+    # download weights
+    if pretrained and weights_url:
+        if checkpoints_path:
+            save_dir = checkpoints_path.model_name
```

Comment:
should this instead be `save_dir = Path(checkpoints_path) / model_name?`

Reply:
Yes but removed because of the subsequent comment